### PR TITLE
IMP-1953 Add normalization models for Primo searches

### PR DESCRIPTION
**Pull request by jazairi**
#### Why these changes are being introduced:

The app can currently search Primo and return raw results, but it cannot
normalize those results and map them into our Result model.

#### Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/IMP-1953

#### How this addresses that need:

This adds the relevant normalization models for Primo to map Primo search
results (both Alma and CDI) into our Result model to render in the result
partial. Primo searches can now be seen in the search_boxed view using
the 'alma' and 'cdi' targets.

#### Side effects of this change:

* Ex Libris has minimal useful documentation on the Primo Search API, so
this code will likely need to be refined over time.
* In an effort to keep things as simple as possible, this changeset
largely mirrors the logic for EDS normalization and minimizes changes
to the Result model and corresponding views. Because of this, some of the
code is a little more convoluted than if we were starting from scratch.
* We are constructing thumbnail URLs to Syndetics using the ISBN. A
PNX record can have multiple ISBNs, so we are taking the first entry
for now.
* All PNX fields are returned as arrays, so it is unclear which are
supposed to be multi-value and which are single-value. We are making some
assumptions to this effect.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

**Comment by jazairi** [comment]
I'll leave the Code Climate issues open pending code review. One is due to excessive cognitive complexity, and the others are for not being DRY. The lack of DRYness is by design, but I'm happy to reconsider if the reviewer feels it should change.

File: app/models/normalize_primo_articles.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    result
+  end
+
+  def journal_title
+    return unless @record['pnx']['addata']['jtitle']
+    @record['pnx']['addata']['jtitle'].join('')
+  end
+
+  def numbering
+    return unless @record['pnx']['addata']['volume'] && @record['pnx']['addata']['issue']
```
Comment (JPrevost) [inline app/models/normalize_primo_articles.rb:20]: I don't have an example record, but I assume some journals have volumes but not issues.

I'm not opposed to leaving this as is with an additional ticket to return to it though.

File: app/models/normalize_primo_books.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  def book_metadata(result)
+    result.thumbnail = thumbnail
+    result.publisher = publisher
+    result.location = location
+    result.subjects = subjects
+    result
+  end
+
+  def thumbnail
+    return unless @record['pnx']['addata']['isbn']
```
Comment (JPrevost) [inline app/models/normalize_primo_books.rb:17]: We may want to consider using issn on the articles normalizer to add thumbnails as well. (not asking for a change to this PR, just noting I think syndetics can also accept issn and may be worth asking UX if they want that)

**Review by JPrevost — APPROVED** [review]
👍🏻 

I won't fixed a few code climate and accepted a couple to return to later.

My comments are non-blocking on this work. I do think we should open new tickets for the volume / issue question and the syndetics for journals question.

File: app/models/normalize_primo_articles.rb (hunk 1 above)
Comment (jazairi) [reply app/models/normalize_primo_articles.rb:20]: Oh, that's a good point. I'll refactor to account for that.

File: app/models/normalize_primo_books.rb (hunk 2 above)
Comment (jazairi) [reply app/models/normalize_primo_books.rb:17]: Good call. I'll make a note in the QA ticket.

**Review by JPrevost — APPROVED** [review]

**Comment by jazairi** [comment]
As per our usual practice, stakeholder review will occur in staging.

**Pull request merged**